### PR TITLE
Updated GetMarsRoverPhotosByDateAndRover method to allow optional rov…

### PR DESCRIPTION
**Pull request by Ray-Sewell**
…er + camera parameters, removed generic call

### CODE REVIEW

File: MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs [hunk 1]
```diff
@@ -11,28 +11,36 @@ namespace MarsRoverBestGroup3._0
...
         }
-        public static List<Photo> GetMarsRoverPhotosByDate(DateTime earth_date)
+        public static List<Photo> GetMarsRoverPhotosByDateAndRover(DateTime earth_date, string rover_name = "curiosity", string camera_name = null)
         {
             var earth_date_to_string = earth_date.ToString("yyyy-MM-dd");
-            var response = GenericCall($"/mars-photos/api/v1/rovers/curiosity/photos?earth_date={earth_date_to_string}&");
+            var api_key = ConfigurationManager.AppSettings["APIKey"];
+            var client = new RestClient("https://api.nasa.gov");
+            var request = new RestRequest($"/mars-photos/api/v1/rovers/{rover_name}/photos", DataFormat.Json);
+            request.AddParameter("earth_date", earth_date_to_string);
```
Comment (MichaelWalker) [inline MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs:33]: Should this be `AddQueryParameter` rather than just `AddParameter`?

Comment (MichaelWalker) [reply MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs:33]: (similarly for all the other ones)

File: MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs [hunk 2]
```diff
@@ -11,28 +11,36 @@ namespace MarsRoverBestGroup3._0
...
+            {
+                request.AddParameter("camera", camera_name);
+            }
+            request.AddParameter("api_key", api_key);
+            var response = client.Get(request);
+            if (response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("API call invalid");
+            }
             var mars_rover_pictures = JsonConvert.DeserializeObject<MarsRoverPhotosResponse>(response.Content);
```
Comment (MichaelWalker) [inline MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs:44]: you should be able to get RestSharp to do this for you.

```csharp
var response = client.Get<MarsRoverPhotosResponse>(request);
return response.Data.Photos;
```

Comment (Ray-Sewell) [reply MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs]: Managed to get this working with the GetMarsRoverPhotosByDateAndRover method, although no luck with the AstronomyPhotoOfTheDay method

File: MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs [hunk 3]
```diff
@@ -11,28 +11,36 @@ namespace MarsRoverBestGroup3._0
...
+            if (camera_name is not null)
+            {
+                request.AddParameter("camera", camera_name);
+            }
+            request.AddParameter("api_key", api_key);
+            var response = client.Get(request);
+            if (response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("API call invalid");
+            }
```
[the remark this answers is not in the document]
Comment (MichaelWalker) [reply MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs]: You seem to have deleted this code now - was that deliberate?
Feels like this error handling code might still be useful?

**Pull request merged**